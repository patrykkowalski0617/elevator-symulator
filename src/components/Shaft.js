import React from "react";
import styled from "styled-components";
import Car from "./Car";

export const ShaftStyled = styled.div`
    background: #777;
    width: ${props => props.carWidth}%;
    position: relative;
    display: flex;
    align-items: flex-end;
    &::before,
    &::after {
        display: block;
        position: absolute;
        height: 2px;
        width: 100%;
        background: #222;
        content: "";
    }
    &::after {
        bottom: 0;
    }
    &::before {
        top: 0;
    }
`;

const Shaft = ({ numberOfCars, carWidth, numberOfFloors }) => {
    const renderCars = () => {
        const arr = [];
        for (let i = 0; i < numberOfCars; i++) {
            arr.push(
                <ShaftStyled key={i} carWidth={carWidth}>
                    <Car carId={i} numberOfFloors={numberOfFloors} />
                </ShaftStyled>
            );
        }

        return arr;
    };

    return <>{renderCars()}</>;
};

export default Shaft;
