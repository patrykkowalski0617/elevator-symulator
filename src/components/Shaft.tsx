import React from "react";
import styled from "styled-components";
import Car from "./Car";

export const ShaftStyled = styled.div<{ carWidth: string }>`
    background-color: #778;
    width: ${props => props.carWidth};
    position: relative;
    display: flex;
    align-items: flex-end;
    &::before,
    &::after {
        display: block;
        position: absolute;
        height: 2px;
        width: 100%;
        background: #222;
        content: "";
    }
    &::after {
        bottom: 0;
    }
    &::before {
        top: 0;
    }
`;

type ShaftProps = {
    numberOfCars: number;
    carWidth: string;
    numberOfFloors: number;
};

const Shaft = ({ numberOfCars, carWidth, numberOfFloors }: ShaftProps) => {
    const renderCars = () => {
        const arr: JSX.Element[] = [];
        for (let i = 0; i < numberOfCars; i++) {
            arr.push(
                <ShaftStyled key={i} carWidth={carWidth}>
                    <Car carId={i} numberOfFloors={numberOfFloors} />
                </ShaftStyled>
            );
        }

        return arr;
    };

    return <>{renderCars()}</>;
};

export default Shaft;
