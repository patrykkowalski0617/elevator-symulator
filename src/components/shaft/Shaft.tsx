import React from "react";
import { ShaftStyled } from "./ShaftStyled";
import Car from "../car/Car";

type ShaftProps = {
    numberOfCars: number;
    carWidth: string;
    numberOfFloors: number;
};

const Shaft = ({ numberOfCars, carWidth, numberOfFloors }: ShaftProps) => {
    const renderCars = () => {
        const arr: JSX.Element[] = [];
        for (let i = 0; i < numberOfCars; i++) {
            arr.push(
                <ShaftStyled key={i} carWidth={carWidth}>
                    <Car carId={i} numberOfFloors={numberOfFloors} />
                </ShaftStyled>
            );
        }

        return arr;
    };

    return <>{renderCars()}</>;
};

export default Shaft;
