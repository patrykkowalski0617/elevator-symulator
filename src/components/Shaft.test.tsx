import React from "react";
import Shaft, { ShaftStyled } from "./Shaft";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";

test("Shaft - render and match snapshot", () => {
    const tree = renderer.create(<Shaft></Shaft>);
    expect(tree.toJSON()).toMatchSnapshot();
});

afterEach(cleanup);
test("Shaft has no vertical border, padding and margin", () => {
    render(<ShaftStyled carWidth={10}></ShaftStyled>);

    const className = ShaftStyled.styledComponentId;
    const ShaftRoots = document.getElementsByClassName(className)[0];
    const style = window.getComputedStyle(ShaftRoots);

    expect(style.borderTopWidth).toBe("");
    expect(style.borderBottomWidth).toBe("");
    expect(style.paddingTop).toBe("");
    expect(style.paddingBottom).toBe("");
});
