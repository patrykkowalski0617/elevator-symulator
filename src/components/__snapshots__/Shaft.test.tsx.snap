// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shaft - render and match snapshot 1`] = `null`;
